compiler/src/dmd/func.d

```diff
@@ -2924,89 +2924,100 @@ Expression addInvariant(AggregateDeclaration ad, VarDeclaration vthis)
  */
 extern (D) int overloadApply(Dsymbol fstart, scope int delegate(Dsymbol) dg, Scope* sc = null)
 {
-    Dsymbol next;
-    for (auto d = fstart; d; d = next)
+    Dsymbols visited;
+
+    int overloadApplyRecurse(Dsymbol fstart, scope int delegate(Dsymbol) dg, Scope* sc)
     {
-        import dmd.access : checkSymbolAccess;
-        if (auto od = d.isOverDeclaration())
+        // Detect cyclic calls.
+        if (visited.contains(fstart))
```

Comment:
Usually you'd use an associative array for this, I suppose overload sets don't get very large.

Reply:
Indeed.  In druntime, it didn't go beyond 4, and in the testsuite the maximum number of recursions was 5 (runnable/ufcs.d).

When building druntime:
```
  40853 visited = 1
     38 visited = 2
    112 visited = 3
     64 visited = 4
```
Because array sizes of 1 are on the stack, makes it cheap for the 99% of cases vs. AA which I'd expect to always be heap.